enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
+
+  def my_all?
+    to_a.my_each { |item| return false if yield(item) == false }
+    true
+  end
+
+  def my_any?
+    to_a.my_each { |item| return true if yield(item) }
+    false
+  end
+
+  def my_none?
+    to_a.my_each { |item| return false if yield(item) }
+    true
+  end
+
+  def my_count?
+    i = 0
+    to_a.my_each { |_item| i += 1 }
+    i
+  end
+
+  def my_map(proc = nil)
+    new_array = []
+    if proc.nil?
+      to_a.my_each { |item| new_array << yield(item) }
+
+    else
+      to_a.my_each { |item| new_array << proc.call(item) }
+    end
+    new_array
+  end
+
+  def my_inject
+    sum = 0
+    to_a.my_each { |item| sum = yield(sum, item) }
+    return sum if sum != 0
+
+    sum = 1
+    to_a.my_each { |item| sum = yield(sum, item) }
+    sum
+  end
```

Comment:
- [ ] enumerables #my_inject searches for the longest word in an array of strings
      Failure/Error: let(:search) { proc { |memo, word| memo.length > word.length ? memo : word } }

- [ ] enumerables #my_inject when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }

- [ ] enumerables #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)

- [ ]  enumerables #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
      Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)

- [ ] enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(20, :*)).to eq array.inject(20, :*)

- [ ] enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range
      Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)